codecarbon/emissions_tracker.py

```diff
@@ -203,13 +212,13 @@ def _measure_power(self) -> None:
         self._last_measured_time = time.time()
 
 
+@suppress(Exception)
 class OfflineEmissionsTracker(BaseEmissionsTracker):
     """
     Offline implementation of the `EmissionsTracker`
     In addition to the standard arguments, the following are required.
     """
-
-    @suppress(Exception)
```

Comment:
This line should be left as it is

Reply:
@LiamConnell that's very strange, I didn't add any suppression :-(, I did however sync my branch with the upstream before adding your previous suggestion....I will remove it and push again, cheers!

Reply:
@LiamConnell Ok now I understand what's happening - there is lot of back-and-forth happening here. In the original codecarbon codebase that I forked, the exception suppression was already there. My original solution was to actually handle unsupported/lack of hardware via ValueError exception, and I removed exception suppression. @kngoyal then requested I put the suppression back and instead handle this issue via a warning. So I put the suppression back and implemented the warning. Now I've been requested to remove the suppression :-o.

I think there needs to be some agreement regarding certain design patterns, such as exception handling. However I don't think this should be tied to this PR. I've implemented an explicit warning for the issue raised, general exception handling and suppression should be handled somewhere else, when a consensus/general agreement is reached in the project.

Reply:
I think I understand where the confusion comes from: It comes from _where_ you added the suppression back in. You added it as a class decorator rather than a method decorator. It only works as a method decorator. 

There is agreement on the design pattern: all public methods of the EmissionsTracker objects have exceptions suppressed. The reasoning is that this library should NEVER cause the user's code to fail, even if the library fails to track carbon. 

You still need to add this line back in, since the method must be decorated.

Reply:
I see @LiamConnell , thanks for the clarification and apologies for the confusion! I have put the suppression back on the constructor and pushed the changes. Phew!